src/mutations/creators.ts

```diff
@@ -53,24 +54,46 @@ export const createTypeAdditionMutation = (
 		) ||
 		isKnownGlobalBaseType(declaredType)
 	) {
-		return {
-			insertion: ` ${newTypeAlias}`,
-			range: {
-				begin: node.type.pos,
-				end: node.type.end,
-			},
-			type: "text-swap",
-		};
+		return textSwap(` ${newTypeAlias}`, node.type.pos, node.type.end);
+	}
+
+	const isAsyncFunction =
```

Comment:
[Refactor] 🤔 `createTypeAdditionMutation` is meant to be a pretty generalized utility. It doesn't understand anything about the types: just that there's a single `declaredType` and an array of `allAssignedTypes`. I don't think this is the right place to add logic specific to async functions.

Given that this PR is only meant to touch `fixIncompleteReturnTypes`, I think the logic should exist there. That would be a place to put logic around promises.